crates/engine/tree/src/static_files.rs

```diff
@@ -0,0 +1,246 @@
+use reth_prune::PruneModes;
+use std::sync::{
+    mpsc::{Receiver, SendError, Sender},
+    Arc,
+};
+use tokio::sync::oneshot;
+
+use crate::{
+    persistence::{PersistenceAction, PersistenceHandle},
+    tree::ExecutedBlock,
+};
+
+/// Writes finalized blocks to reth's static files.
+///
+/// This is meant to be a spawned task that listens for various incoming finalization operations,
+/// and writing to or producing new static files.
+///
+/// This should be spawned in its own thread with [`std::thread::spawn`], since this performs
+/// blocking file operations in an endless loop.
+#[derive(Debug)]
+pub struct StaticFileTask<DB> {
+    /// The db / static file provider to use
+    provider: ProviderFactory<DB>,
+    /// Handle for the database task
+    database_handle: PersistenceHandle,
+    /// Incoming requests to write static files
+    incoming: Receiver<StaticFileAction>,
+    /// The pruning configuration
+    pruning: PruneModes,
+}
+
+impl<DB> StaticFileTask<DB>
+where
+    DB: Database + 'static,
+{
+    /// Create a new static file task, spawning it, and returning a [`StaticFileTaskHandle`].
+    fn spawn_new(provider: ProviderFactory<DB>) -> StaticFileTaskHandle {
+        todo!("implement initialization first");
+        // let (tx, rx) = std::sync::mpsc::channel();
+        // let task = Self::new(provider, rx);
+        // std::thread::Builder::new()
+        //     .name("StaticFile Task".to_string())
+        //     .spawn(|| task.run())
+        //     .unwrap();
+
+        // StaticFileTaskHandle::new(tx)
+    }
+
+    // TODO: some things about this are a bit weird, and just to make the underlying static file
+    // writes work - tx number, total difficulty inclusion. They require either additional in memory
+    // data or a db lookup. Maybe we can use a db read here
+    /// Writes the transactions to static files, to act as a log.
+    ///
+    /// This will then send a command to the db task, that it should update the checkpoints for
+    /// headers and block bodies.
+    fn log_transactions(
+        &self,
+        block: Arc<SealedBlock>,
+        start_tx_number: u64,
+        td: U256,
+        sender: oneshot::Sender<()>,
+    ) -> ProviderResult<()> {
+        let provider = self.provider.static_file_provider();
+        let mut header_writer = provider.get_writer(block.number, StaticFileSegment::Headers)?;
+        let mut transactions_writer =
+            provider.get_writer(block.number, StaticFileSegment::Transactions)?;
+
+        // TODO: does to_compact require ownership?
+        header_writer.append_header(block.header().clone(), td, block.hash())?;
+        let no_hash_transactions =
+            block.body.clone().into_iter().map(TransactionSignedNoHash::from);
+
+        let mut tx_number = start_tx_number;
+        for tx in no_hash_transactions {
+            transactions_writer.append_transaction(tx_number, tx)?;
+            tx_number += 1;
```

Comment:
where is this `tx_number` used?

Reply:
for `append_transaction` above it, it's required by the static file `append_transaction` method

Reply:
🙄 absolutely, sorry missed it